src/library_pthread.js

```diff
@@ -747,7 +755,6 @@ var LibraryPThread = {
   },
 
   // Public pthread_self() function which returns a unique ID for the thread.
-  pthread_self__asm: true,
```

Comment:
Instead of removing the `__asm`, could this also add `pthread_self__sig: 'i'` if the absence of that ran into errors like you described in the comments? It is important for performance to avoid having to do an FFI here, because `pthread_self()` can be called a lot in performance intensive code.